src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -181,6 +199,75 @@ private static uint HandleEventStreamsAvailable(State state, ref ConnectionEvent
             return MsQuicStatusCodes.Success;
         }
 
+        private static uint HandleEventPeerCertificateReceived(State state, ref ConnectionEvent connectionEvent)
+        {
+            SslPolicyErrors sslPolicyErrors  = SslPolicyErrors.None;
+            X509Chain? chain = null;
+            X509Certificate2? certificate = null;
+
+            if (!OperatingSystem.IsWindows())
+            {
+                // TODO fix validation with OpenSSL
```

Comment:
What is remaining to do here? Is there an issue?

Reply:
The issue is that we don't have good way how to get certificate chain right now with OpenSSL. 
There is not specific issue AFAIK. I mentally do this as contributions to  #49574.
I will follow-up if/when https://github.com/microsoft/msquic/pull/1450 (+implementation) is ready.